src/nupic/algorithms/Cells4.cpp

```diff
@@ -188,16 +188,14 @@ bool Cells4::computeUpdate(UInt cellIdx, UInt segIdx, CStateIndexed& activeState
       highWaterSize = segment.size();
       newSynapses.reserve(highWaterSize);
     }
-    for (UInt i = 0; i != segment.size(); ++i)
+    for (UInt i=0; i<segment.size(); ++i) //FIXME why we skip segment[0] here?
```

Comment:
The first iteration of the loop will be `i=0`. I prefer `++i` over `i++`. They have the same effect if they are used alone. When they are used as part of a more complex statement then `++i` will increment and evaluate to the incremented value while `i++` will increment the variable but evaluate to the pre-incremented value.

So I like switching to `i < segment.size()` (I'd prefer spaces around the comparison operator though), don't have a strong opinion on the `++i` vs `i++` but regardless the comment needs to go.